src/Interpreters/Context.cpp

```diff
@@ -2128,7 +2128,12 @@ zkutil::ZooKeeperPtr Context::getZooKeeper() const
 
     const auto & config = shared->zookeeper_config ? *shared->zookeeper_config : getConfigRef();
     if (!shared->zookeeper)
-        shared->zookeeper = std::make_shared<zkutil::ZooKeeper>(config, "zookeeper", getZooKeeperLog());
+        shared->zookeeper = std::make_shared<zkutil::ZooKeeper>(
+            config,
+            config.has("zookeeper")    ? "zookeeper"
+                : config.has("keeper") ? "keeper"
+                                       : "keeper_server",
```

Comment:
Please create a function for this.
You can also store in `std::array`, iterate and return first contained in config.